ui/redux/actions/file.js

```diff
@@ -111,86 +111,81 @@ export function doDeleteFileAndMaybeGoBack(
   };
 }
 
-export function doFileGet(uri: string, saveFile: boolean = true, onSuccess?: (GetResponse) => any) {
-  return async (dispatch: Dispatch, getState: () => any) => {
-    const state = getState();
-    let claim = selectClaimForUri(state, uri);
-    if (!claim) {
-      claim = await dispatch(doGetClaimFromUriResolve(uri));
-    }
-    const isLivestreamClaim = isStreamPlaceholderClaim(claim);
-    const { nout, txid } = claim;
-    const outpoint = `${txid}:${nout}`;
+export const doFileGetForUri = (uri: string, onSuccess?: (GetResponse) => any) => async (
+  dispatch: Dispatch,
+  getState: GetState
+) => {
+  const state = getState();
+  const alreadyFetching = selectOutpointFetchingForUri(state, uri);
 
-    dispatch({
-      type: ACTIONS.FETCH_FILE_INFO_STARTED,
-      data: {
-        outpoint,
-      },
-    });
+  if (alreadyFetching) return;
 
```

Comment:
:small_blue_diamond: Double-check: what happens to a caller with `onSuccess` when there is a pending fetch?

Not sure if it will impact `doPlayUri --> doPurchaseUri`, but from a function point of view, unclear what's the expectation for all scenarios.

- Caller with callback:
    - Pass:  callback runs +  `FETCH_FILE_INFO_COMPLETED`
    - Fail:  `FETCH_FILE_INFO_FAILED`
    - Fetching:  ?? (dead end, need to listen to other selectors)

Maybe not a concern, but just double-checking.  Haven't tried making an LBC purchase.

Reply:
Good point, 5042117bce37d213e10f19976841dfabd060719c should be enough but there might be a better approach without calling again while pending